### ISSUE TITLE
Minor optimization for SQLServerProvider translating IsNullOrWhiteSpace as LTRIM(RTRIM(..))

### ISSUE THREAD

**Issue by CaiusJard** [dotnet/efcore#26236]
Documentation here https://docs.microsoft.com/en-us/ef/core/providers/sql-server/functions

says that INOW() translates as 

    @value IS NULL OR LTRIM(RTRIM(@value)) = N''
    
If a string is whitespace, to my mind RTRIM would have reduced it to `''` already, so the decision can be made as to whether it's WS without the need to call LTRIM on it..?

**Comment by roji** [comment]
In fact, since SQL Server disregards all trailing whitespace when comparing strings, we should simply be able to translate to:

```sql
@value IS NULL OR @value = N''
```

This should even be sargable...

**Comment by stevendarby** [comment]
Is this fixed in 6.0? https://github.com/dotnet/efcore/issues/22916

**Comment by roji** [comment]
@stevendarby good catch - [this is indeed already fixed in 6.0](https://github.com/dotnet/efcore/blob/release/6.0/src/EFCore.SqlServer/Query/Internal/SqlServerStringMethodTranslator.cs#L353).

Submitted #3461 to update.
